src/plot_api/plot_api.js

```diff
@@ -2913,10 +2966,12 @@ function getDiffFlags(oldContainer, newContainer, outerparts, opts) {
 
     for(key in oldContainer) {
         // short-circuit based on previous calls or previous keys that already maximized the pathway
-        if(flags.calc) return;
+        if(flags.calc && !opts.transition) return;
```

Comment:
Can we move these tests up or down?

Reply:
Why?

Reply:
First I thought it may be independent from the `key` and loop.
Resolved.